### PR TITLE
Settings Page

### LINKED ISSUE #8: Settings Page
Add a Settings page. This will be where the user goes to accomplish various backend related tasks, including possibly:

- [ ] import/export JSON
- [ ] import/export to cloud services
- [ ] reset walkthrough
- [ ] show/hide help text
- [ ] save settings to Auth0 user profile
- [ ] sync options (online/offline, cloud services, DLx database)
  - This could just be a checklist of the places you'd like your data to sync
  - Checking no options means that your data is only saved locally

### LINKED ISSUE #242: move page-specific modules to live within their page folders
Page-specific modules should live within their page folders.

The `/components` folder should be reserved for components that are in fact reused across pages.

For example:

```
/app
  /Banner
  /Nav
  /Status
/pages
  /languages
    /LanguageEditor
    /LanguagesList
```

### LINKED ISSUE #253: update DLx JavaScript library to v0.3.0
Specifically, we need a version of the library which includes `iso` and `glottolog` properties on the `Language` model.

- [x] Update DLx JavaScript library
- [x] Remove ISO Code code in `src/models/Language.js`

This will allow the testing data to be formatted correctly.

### PR DESCRIPTION
**Pull request by dwhieb**
* Creates a skeleton Settings Page, with HTML and styling. (closes #8)
* Reorganizes the project so that components live under their relevant page. (closes #242)
* Updates DLx JS library. (closes #253)

### CODE REVIEW

**Comment by dwhieb** [comment]
[[app] Settings Page](https://trello.com/c/UN1eDQMq/262-app-settings-page)

File: .github/MAINTAINERS.md [hunk 1]
```diff
@@ -94,7 +94,7 @@ These are the files and folders located in the `/src` folder.
 
 Folder / File | Description
 ------------- | -----------
-`/components` | Each folder contains all the code for a single, reusable component, including HTML, LESS, JS, and tests.
```
Comment (vadekh) [inline .github/MAINTAINERS.md:97]: I understand the possible future uses of this path, but it doesnt seem to exist anymore in this branch so is this description required?

File: src/app/Nav/Nav.test.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    cy.get(`#nav a[href="#/languages"]`).click();
+    cy.get(`#main[data-page=languages]`);
+    // current page should be set
+    cy.get(`#nav a[href="#/languages"]`)
+    .should(`have.attr`, `aria-describedby`, `current-page`)
+    .should(`have.class`, `current`);
+
+    cy.get(`#nav a[href="#/settings"]`).click();
+    cy.get(`#main[data-page=settings]`);
+    // previous page should be unset
```
Comment (vadekh) [inline src/app/Nav/Nav.test.js:21]: Shouldn't we add one more check here to ensure that `settings` is the new current page? Other than that it looks good.

**Comment by vadekh** [comment]
Here are the visual errors mentioned above:
Screen Shot 2020-08-08 at 6 31 56 PM

**Comment by dwhieb** [comment]
Yeah the components have all been moved to their respective pages (see #242), which this PR also closes. I'm not sure why they're not rendering though.

**Comment by vadekh** [comment]
@dwhieb I know the big ones were moved, but what about Banner and Link and Skip Link etc? The files all say the were deleted not moved in this PR

**Comment by dwhieb** [comment]
I moved those into the App view, either into App.less or the main index.html. The Nav is now a component in the /app folder. There was no reason to have them as separate components because they're only used once.

**Comment by vadekh** [comment]
@dwhieb Ok yes I see them now!

File: .github/MAINTAINERS.md (hunk 1 above)
Comment (dwhieb) [reply .github/MAINTAINERS.md:97]: Agreed. We can add this again whenever we have a component that gets used across pages.

File: src/app/Nav/Nav.test.js (hunk 2 above)
Comment (dwhieb) [reply src/app/Nav/Nav.test.js:21]: Yup - good call!

**Comment by dwhieb** [comment]
@vadekh I couldn't reproduce the visual errors. Are you still getting this? Does it display correctly when you run the Cypress tests manually? How about with an incognito window?

**Comment by vadekh** [comment]
* @dwhieb no longer getting the visual errors!
* Also the nav.test now has `cy.get(#main[settings])` twice in a row, and I'm not sure if this now checks that settings is the current page. Again, I assume that it works, but for consistency we should check that switching works. I can adjust the test this afternoon.
* Lastly, the `button red` styling doesn't seem to be working on the delete language button now. I know the styling stuff has been moved around a bit, not sure where the break might be.

**Comment by dwhieb** [comment]
Great! Don't worry about the button styling for this PR. There's another issue for that. That styling belongs in LanguageEditor.less, but we haven't set up that file yet.

**Comment by vadekh** [comment]
@dwhieb double check my change to `Nav.test.js` and then its good to go!

**Pull request merged**